cpp/tests/bitmask/bitmask_tests.cpp

```diff
@@ -384,8 +384,12 @@ void cleanEndWord(rmm::device_buffer &mask, int begin_bit, int end_bit)
   auto number_of_mask_words = cudf::num_bitmask_words(static_cast<size_t>(end_bit - begin_bit));
   auto number_of_bits       = end_bit - begin_bit;
   if (number_of_bits % 32 != 0) {
-    auto end_mask = ptr[number_of_mask_words - 1];
-    end_mask      = end_mask & ((1 << (number_of_bits % 32)) - 1);
+    cudf::bitmask_type end_mask = 0;
+    CUDA_TRY(cudaMemcpy(
+      &end_mask, ptr + number_of_mask_words - 1, sizeof(end_mask), cudaMemcpyDeviceToHost));
+    end_mask = end_mask & ((1 << (number_of_bits % 32)) - 1);
+    CUDA_TRY(cudaMemcpy(
+      ptr + number_of_mask_words - 1, &end_mask, sizeof(end_mask), cudaMemcpyHostToDevice));
```

Comment:
This doesn't really matter since it's a test, but could also use this:
```suggestion
    cudf::bitmask_type end_mask = 0;
    cudf::detail::device_single_thread(
      [ptr, index = number_of_mask_words - 1, number_of_bits] __device__() {
        ptr[index] = ptr[index] & ((1 << (number_of_bits % 32)) - 1);
      }, rmm::cuda_stream_default);
    rmm::cuda_stream_default.synchronize();
```

Reply:
Introducing a device lambda here would require changing this to a `.cu` file. 
It did not seem worth it since this function is only used a few times.

Reply:
Ah, didn't think of that.